subprojects/scala/src/test/groovy/org/gradle/api/tasks/scala/ScalaDocSpec.groovy

```diff
@@ -16,21 +16,40 @@
 package org.gradle.api.tasks.scala
 
 import org.gradle.api.internal.AbstractTask
-import org.gradle.api.tasks.AbstractSpockTaskTest
+import org.gradle.api.internal.file.TestFiles
+import org.gradle.api.internal.project.taskfactory.TaskInstantiator
+import org.gradle.api.tasks.AbstractConventionTaskTest
+
 import org.gradle.util.WrapUtil
 
 import static org.gradle.api.tasks.compile.AbstractCompileTest.*
 
-class ScalaDocSpec extends AbstractSpockTaskTest {
+class ScalaDocSpec extends AbstractConventionTaskTest {
+
     ScalaDoc scalaDoc
 
+    def setup() {
+        scalaDoc = project.services.get(TaskInstantiator).create(TEST_TASK_NAME, ScalaDoc)
```

Comment:
Hopefully this is the desired way to instantiate the task.

I found that `AbstractSpockTaskTest` had also limitations on this by doing https://github.com/gradle/gradle/blob/master/subprojects/core/src/testFixtures/groovy/org/gradle/api/tasks/AbstractSpockTaskTest.groovy#L55-L59